akka-http-core/src/main/scala/akka/http/javadsl/ServerBinding.scala

```diff
@@ -26,5 +30,46 @@ class ServerBinding private[http] (delegate: akka.http.scaladsl.Http.ServerBindi
    *
    * The produced [[java.util.concurrent.CompletionStage]] is fulfilled when the unbinding has been completed.
    */
-  def unbind(): CompletionStage[Done] = delegate.unbind().toJava
+  def unbind(): CompletionStage[Done] =
+    delegate.unbind().toJava
+
+  // TODO docs after review
+  def terminate(hardDeadline: java.time.Duration): CompletionStage[HttpTerminated] = {
+    delegate.terminate(FiniteDuration.apply(hardDeadline.toMillis, TimeUnit.MILLISECONDS))
```

Comment:
didn't we have a `toScala` somewhere for JDuration->FiniteDuration?

Reply:
thanks!